src/Raven.Studio/typescript/components/pages/database/documents/identities/DocumentIdentitiesAddModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { ModalProps } from "reactstrap/types/lib/Modal";
+import { Button, InputGroup, Label, Modal, ModalBody, ModalFooter } from "reactstrap";
+import React, { useEffect } from "react";
+import { Icon } from "components/common/Icon";
+import { FormInput } from "components/common/Form";
+import { Control, SubmitHandler, useForm, useWatch } from "react-hook-form";
+import classNames from "classnames";
+import {
+    AddIdentitiesFormData,
+    addIdentitiesYupResolver,
+} from "components/pages/database/documents/identities/DocumentIdentitiesValidation";
+import ButtonWithSpinner from "components/common/ButtonWithSpinner";
+import { useServices } from "hooks/useServices";
+import { useAppSelector } from "components/store";
+import { databaseSelectors } from "components/common/shell/databaseSliceSelectors";
+import { tryHandleSubmit } from "components/utils/common";
+import RichAlert from "components/common/RichAlert";
+import { useAsync } from "react-async-hook";
+import { LazyLoad } from "components/common/LazyLoad";
+import { isFunction } from "lodash";
+
+interface DocumentIdentitiesAddModalProps extends ModalProps {
+    toggleModal: () => void;
+    defaultValues?: AddIdentitiesFormData;
+    identities?: AddIdentitiesFormData[];
+    refetch?: () => void;
+}
+
+export default function DocumentIdentitiesAddModal({
+    defaultValues,
+    contentClassName,
+    wrapClassName,
+    refetch,
+    identities,
+    ...props
+}: DocumentIdentitiesAddModalProps) {
+    const databaseName = useAppSelector(databaseSelectors.activeDatabaseName);
+    const { databasesService } = useServices();
+    const form = useForm<AddIdentitiesFormData>({
+        resolver: addIdentitiesYupResolver,
+        defaultValues,
+    });
+
+    const formValues = useWatch({ control: form.control });
+
+    useEffect(() => {
+        const prefixExists = identities?.some(
+            (identity) => identity.prefix?.toLocaleLowerCase().slice(0, -1) === formValues.prefix?.toLocaleLowerCase()
+        );
+
+        if (prefixExists) {
+            form.setError("prefix", { type: "custom", message: "A prefix with the same name exists" });
+        } else {
+            form.clearErrors("prefix");
+        }
+    }, [formValues, identities, form]);
+
+    const isEditing = !!defaultValues;
+
+    const onSubmit: SubmitHandler<AddIdentitiesFormData> = ({ prefix, value }) => {
+        return tryHandleSubmit(async () => {
+            await databasesService.seedIdentity(databaseName, prefix, value);
+            form.reset();
+            if (isFunction(refetch)) refetch();
```

Comment:
Let's maintain consistency in the codebase and add braces
```suggestion
            if (isFunction(refetch)) {
                refetch();
            }
```

Reply:
Does this mean that the data in the table will not be refreshed after editing?